server/api/include/irods/rs_check_auth_credentials.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef IRODS_RS_CHECK_AUTH_CREDENTIALS_HPP
+#define IRODS_RS_CHECK_AUTH_CREDENTIALS_HPP
+
+/// \file
+
+#include "irods/check_auth_credentials.h"
+
+struct RsComm;
+
+/// Verifies the credentials passed match the credentials stored in the catalog.
+///
+/// This API endpoint only supports Native authentication and requires rodsadmin
+/// privileges.
+///
+/// It does NOT modify anything state on either side of the connection.
+///
+/// \param[in] _comm   A pointer to a RsComm.
+/// \param[in] _input  A pointer to a CheckAuthCredentialsInput.
+/// \param[in] _output A pointer to a CheckAuthCredentialsOutput.
+///
+/// \return An integer.
+/// \retval 0  On success.
+/// \retval <0 On failure.
+///
+/// \b Example
+/// \code{.cpp}
+/// RsComm* comm = // Our rodsadmin connection object.
+///
+/// // Configure the input object for the API call.
+/// struct CheckAuthCredentialsInput input;
+/// memset(&input, 0, sizeof(struct CheckAuthCredentialsInput));
+///
+/// strncpy(input.username, "bob", sizeof(input.username) - 1);
+/// strncpy(input.zone, "tempZone", sizeof(input.zone) - 1);
+///
+/// char* expected_password = // What we expect bob's password to be.
+/// char* scrambled_password = // expected_password, but scrambled. See #obfEncodeByKey.
+/// strncpy(input.password, scrambled_password, sizeof(input.password) - 1);
+///
+/// struct CheckAuthCredentialsOutput* output = NULL;
+///
+/// // Verify if the given password for bob is correct.
+/// const int ec = rs_check_auth_credentials(comm, &input, &output);
+///
+/// if (ec < 0) {
+///     // Handle error.
```

Comment:
Same comment here about mentioning the fact that `output` will only be allocated when `ec >= 0`.

Should one of these Doxygen comments refer to the other, or just have 2 copies? Not sure what our process is for this.

Reply:
No process established yet. We could link the two, but it hasn't become an issue for me to want to investigate that yet. Especially when new API endpoints tend to not change very much once the dust settles.

Thoughts?

Reply:
There is one small difference between the two and that's the wording for the `RxComm` parameter. There's also the case of example code appearing in the documentation. Dealing with the parameter is simple, but dealing with the example will take a little thought.

We should investigate how to do that, but that can happen at any time.

Marking this resolved. Consider opening an issue to investigate that so we don't forget.